pkgs/applications/emulators/yuzu/default.nix

```diff
@@ -1,28 +1,40 @@
-{ branch ? "mainline", libsForQt5, fetchFromGitHub }:
+{ branch ? "mainline"
+, libsForQt5
+, fetchFromGitHub
+}:
+
 let
-  inherit libsForQt5 fetchFromGitHub;
+  # Fetched from https://api.yuzu-emu.org/gamedb, last updated 2022-03-23.
+  # Please make sure to update this when updating yuzu!
+  compat-list = ./compatibility-list.json;
 in {
-  mainline = libsForQt5.callPackage ./base.nix rec {
+  mainline = libsForQt5.callPackage ./generic.nix rec {
     pname = "yuzu-mainline";
-    version = "953";
-    branchName = branch;
+    version = "958";
+
     src = fetchFromGitHub {
       owner = "yuzu-emu";
       repo = "yuzu-mainline";
       rev = "mainline-0-${version}";
-      sha256 = "0p07gybyhr6flzmhz92qlrwcq7l37c2wmcxw8sbrvhj2pgaaw9ic";
+      sha256 = "sha256-gQ/0049e8YcL6rfog36Tl5/WXxgtdcl0CuTj/SUzbxo=";
       fetchSubmodules = true;
     };
+
+    inherit branch compat-list;
   };
-  early-access = libsForQt5.callPackage ./base.nix rec {
+
+  early-access = libsForQt5.callPackage ./generic.nix rec {
     pname = "yuzu-ea";
-    version = "2557";
-    branchName = branch;
+    version = "2585";
+
     src = fetchFromGitHub {
       owner = "pineappleEA";
       repo = "pineapple-src";
       rev = "EA-${version}";
-      sha256 = "013xxgyn8y5fv0xbrm0zfl9xmi0gx4hpflrbjskg1hcvb2bjqyvj";
+      sha256 = "sha256-TTfjJBqbIRsHa8draDpqSvozGbgA5/SPSSHV7IESLEg=";
+      fetchSubmodules = true;
```

Comment:
For whatever reason `nix-prefetch-git` generates a wrong checksum when not using `fetchSubmodules`/`--fetch-submodules`. It doesn't actually have any submodules, this is just done for the update script.

Reply:
Did you try if only leaveDotGit works?

Reply:
I haven't, as I had [some issues](https://github.com/NixOS/nixpkgs/pull/84117#discussion_r411703390) with it when initially adding this package. See also https://github.com/NixOS/nixpkgs/issues/8567.

Reply:
IIRC fetchSubmodules has the same issues.

Reply:
It seems like `leaveDotGit` doesn't fix this, sadly.